source/transactions/tests/unified/mongos-unpin.yml

```diff
@@ -0,0 +1,118 @@
+description: mongos-unpin
+
+schemaVersion: '1.1'
+
+runOnRequirements:
+  - minServerVersion: '4.2'
+    topologies: [ sharded-replicaset ]
+
+createEntities:
+  - client:
+      id: &client0 client0
+      useMultipleMongoses: true
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &database0Name mongos-unpin-db
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collection0Name test
+  - session:
+      id: &session0 session0
+      client: *client0
+
+initialData:
+  - collectionName: *collection0Name
+    databaseName: *database0Name
+    documents: []
+
+_yamlAnchors:
+  anchors:
+    # LockTimeout will cause the server to add a TransientTransactionError label. It is not retryable.
+    &lockTimeoutErrorCode 24
+
+tests:
+  - description: unpin after TransientTransctionError error on commit
+    operations:
+      - &startTransaction
+        name: startTransaction
+        object: *session0
+      - &insertOne
+        name: insertOne
+        object: *collection0
+        arguments:
+          document: { x: 1 }
+          session: *session0
+      - name: targetedFailPoint
+        object: testRunner
+        arguments:
+          session: *session0
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+              failCommands: [ commitTransaction ]
+              errorCode: *lockTimeoutErrorCode
+      - name: commitTransaction
+        object: *session0
+        expectError:
+          # LockTimeout is not retryable, so the commit fails.
+          errorCode: *lockTimeoutErrorCode
+          errorLabelsContain: [ TransientTransactionError ]
+          errorLabelsOmit: [ UnknownTransactionCommitResult ]
+      - &assertNoPinnedServer
+        name: assertSessionUnpinned
+        object: testRunner
+        arguments:
+          session: *session0
+
+  - description: unpin on successful abort
```

Comment:
This is a new requirement in the spec that requires driver changes even for drivers that otherwise follow the spec before. Can you call this change out in the spec changelog and bump the version number?

Reply:
Done.